libs/types-common/src/Data/Id.hs

```diff
@@ -317,9 +344,10 @@ instance DecodeWire RequestId where
 
 newtype IdObject a = IdObject {fromIdObject :: a}
   deriving (Eq, Show, Generic)
+  deriving (ToJSON, FromJSON) via Schema (IdObject a)
 
-instance FromJSON a => FromJSON (IdObject a) where
-  parseJSON = withObject "Id" $ \o -> IdObject <$> (o .: "id")
-
-instance ToJSON a => ToJSON (IdObject a) where
-  toJSON (IdObject a) = object ["id" .= a]
+instance ToSchema a => ToSchema (IdObject a) where
+  schema =
+    object "Id" $
+      IdObject
+        <$> fromIdObject .= field "id" schema
```

Comment:
I'm not sure what you mean. Are you saying we should have a more general `ToSchema` instance here? How would that work?

Reply:
I actually don't know how this could work with `ToSchema`. We may have had some values in the code which are not owned by us and so have no `ToSchema` instances. Maybe it is fine, and we can wait for such problems to occur. So, ignore me here, I guess.

Reply:
We can have a combinator version of this instance:
```haskell
idObjectSchema :: ValueSchema d a -> ValueSchema d (IdObject a)
```
then if you need a schema for some `IdObject a` such that `a` is not an instance of `ToSchema`, you can just use that, and provide the schema for `a` by hand. For example, if `a` has JSON and Swagger instances, one can use
```haskell
idObjectSchema genericToSchema
```

Reply:
Sure, let's do that when we need it though.